app/store/persistent-state/index.tsx

```diff
@@ -2,22 +2,27 @@ import { createContext, useContext, PropsWithChildren } from "react"
 import * as React from "react"
 
 import { loadJson, saveJson } from "@app/utils/storage"
+import KeyStoreWrapper from "@app/utils/storage/secureStorage"
 
 import {
   defaultPersistentState,
-  migrateAndGetPersistentState,
+  migrateAndGetLocalStorageState,
   PersistentState,
 } from "./state-migrations"
 
 const PERSISTENT_STATE_KEY = "persistentState"
 
 const loadPersistentState = async (): Promise<PersistentState> => {
   const data = await loadJson(PERSISTENT_STATE_KEY)
-  return migrateAndGetPersistentState(data)
+  const localStorageState = await migrateAndGetLocalStorageState(data)
```

Comment:
I think as the code is written now, it will log people out of the app. I believe it will load the local storage state which no longer has the token and then will it will load the secure storage stage which should start as empty, as a result a user will be unexpectedly logged out. Not totally sure what the correct solution to this is. Maybe as part of the migrating the local storage state it should also save the token to the secure storage state.

Reply:
That's right. The migration will remove the token. I think it makes sense to save the token in secure storage as part of the migration. I'll add that. Thanks!